723 cypress axe wcag testing

louiseClarkUSDS opened the pull request.

louiseClarkUSDS committed:
adds cypress axe tests

louiseClarkUSDS committed:
fixes image alt text in footer

SteveMattinglyUSDS committed:
Add files via upload

SteveMattinglyUSDS committed:
Add files via upload

SteveMattinglyUSDS committed:
Delete refresh-visuals.png

SteveMattinglyUSDS committed:
Add files via upload

SteveMattinglyUSDS committed:
Add files via upload

SteveMattinglyUSDS reviewed (dismissed):
I don't see any problems here, but I have a few questions or maybe suggestions.
First, did you mean to PR to `staging` rather than `master`?
Second, I wonder why you created a separate file for each of your tests. It works, but putting them in one file would probably be more typical.
Finally, I came up with `cypress/e2e/sitemap.sh`, a script that generates a JSON file of all site pages by processing the sitemap. That way I can test all pages-- even future ones-- without having to code for each. Let me know if you'd like to talk about using that approach.

louiseClarkUSDS committed:
adds readme directions and cypress open script

SteveMattinglyUSDS reviewed (approved):
👍

louiseClarkUSDS committed:
adds accessibility full loop test

SteveMattinglyUSDS commented on cypress/e2e/accessibility/07_accessibility.cy.js:
```diff
@@ -0,0 +1,12 @@
+      // Wait for page to load ...
+      cy.intercept(url).as('getPage');
+      cy.visit(url);
+      cy.checkPageA11y(url);    
```
You want to use a `wait` with the @name of the intercept you are doing, like this:
```
      cy.intercept(url).as('getPage');
      cy.visit(url);
      cy.wait('@getPage').then(() => {
        cy.checkPageA11y(url);
      });
```
This ensures that the page is completely loaded before you run your checks.

louiseClarkUSDS replied:
Perfect! I'll make that change.

SteveMattinglyUSDS committed:
Staging (#825)

* updates news logo color and updates h1 text on project pages

* testing changes to staging

* a test for deployment

* testing staging dep

* Update README.md

* Update home.yml

* Update README.md

* Fix banner images gh pages (#806)

* Fix 404 on banner images for nonprod sites

* Fix typo

* Insert disclaimer for non-production deploys (#805)

* added -i flag to change config file

* Create new pages (#804)

* Create shell of md only file

* Create workflow

* Fix indent

* Fix syntax

* Fix permissions

* Create page mynewpage2

* Add job summary; do cleanup

* Create page mypage3

* Fix summary message

* Create page mypage2

* Improve summary markdown

* Create page mypage1

* Fix permalink frontmatter

* Create page try2

* Fix file write

* Create page try1

* Update try1.md

* Experiment with Liquid in markdown

* Improve URL instructions; cleanup

* Create page apage

* Delete test page

* Improve prompt

---------

Co-authored-by: SteveMattinglyUSDS <SteveMattinglyUSDS@users.noreply.github.com>

* Test for visual regressions (#788)

* Test for visual regressions

* Create workflow

* Background server in workflow

* Write report to pull request comment

* Add error handling

* Temporarily expand trigger for testing

* Squashed commit of the following:

commit 257a8cf0133eb68ca2dd5c64cc38cb8891f2db5a
Merge: f7dddd2 d78d88a
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Tue Feb 14 14:28:08 2023 -0400

    Merge branch 'cvr-test' of https://github.com/usds/website into cvr-test

commit f7dddd2cc884e41e89c7f7acc399fb59a393df02
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Tue Feb 14 14:28:06 2023 -0400

    Visual refresh completed

commit d78d88acb0d277a73df88be6a2640c7ea0870989
Author: SteveMattinglyUSDS <SteveMattinglyUSDS@users.noreply.github.com>
Date:   Tue Feb 14 18:15:06 2023 +0000

    Update base images

commit 3780f6ff848033eb4704f7501308d281d0b7fe77
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Tue Feb 14 14:10:14 2023 -0400

    debug

commit c09404e8fbff3bacedffefa317866e60468da0aa
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Tue Feb 14 13:55:37 2023 -0400

    debug

commit a2ddc3e81ce1977ba7728676f038c66d9f46e86d
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Tue Feb 14 13:52:58 2023 -0400

    debug

commit 633c26bbfa3a07e3aab97dd1cffaca4e37421827
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Tue Feb 14 13:30:45 2023 -0400

    debug

commit 955999da9cca085bb7d901ec35466f999fccd803
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Tue Feb 14 13:28:41 2023 -0400

    debug pr checkout

commit 3c97ee9a596e3146a91a130b8f002a67e880cdaf
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Tue Feb 14 13:17:24 2023 -0400

    fetch-depth: 0

commit 43d450bfd965c5e36db0a6e9a1886ba496fce34d
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Tue Feb 14 12:57:14 2023 -0400

    Add image refresh option

commit 6de720163ba8fb933c6b44223c103f9c7a61c5a5
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 23:33:20 2023 -0400

    debug

commit 7ea1925ea8efd8fa8950cff9ea0fb9d74c9ea8c8
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 23:15:24 2023 -0400

    debug

commit 3a27fab11e6b8c1f0c7a2be9f0847316169b2112
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 23:08:32 2023 -0400

    debug

commit b913e04cb4e73841b544b6c37791a8b03d6103a5
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 22:53:08 2023 -0400

    debug

commit 885f78b7ee1d04955695cdbe8a451a9934c15700
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 22:34:07 2023 -0400

    Fix link

commit 24bc2bf1c4e47dacb00d073db1d75848ac238524
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 22:22:31 2023 -0400

    improve feedback

commit a36c2947fb76c80b1f1896c2c9ddf10351ecd240
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 21:44:23 2023 -0400

    Upload artifact

commit bc1da8bdda9be0b65382a35b39dea4bdfa11e705
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 20:42:57 2023 -0400

    allow-repeats

commit a8ae071bb8e01265dcc4ca6f5646dc0c6b0d625a
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 20:36:34 2023 -0400

    testing

commit 2f117823ac31c7d63a0dae3f5efb74dac448af5e
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 20:02:13 2023 -0400

    Debug breaking build

commit e21c973af53a76928a27c67771d73d2b199a46ef
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 19:53:39 2023 -0400

    Add exit 0

commit 16b6687221baeb448ad25234094e4f5242a7d521
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 19:46:49 2023 -0400

    Drop PR comment approach

commit 10c7fd8656b5c5c429a1af6a00874239e6f96a21
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 19:35:51 2023 -0400

    testing

commit 99ffea54513e127a4773f038c23c631b22e9ccf1
Author: SteveMattinglyUSDS <Stephen.S.Mattingly@omb.eop.gov>
Date:   Mon Feb 13 19:25:26 2023 -0400

    Test error handling

* Reenable all tests

* Use sitemap.xml to drive visual tests

* Remove compare option from workflow_dispatch

* Improve user messaging

* Fix messages

* Improve messaging

* Improve visual test performance (#816)

* Add caching to workflow

* Debug caching

* debug caching

* debug caching

* Decrease scroll time

* debug checkout error

* debug checkout error

* debug checkout

* debug checkout; copy wget fix

* Update base images

---------

Co-authored-by: SteveMattinglyUSDS <SteveMattinglyUSDS@users.noreply.github.com>

* Add visual test, create page details (#814)

---------

Co-authored-by: louiseClarkUSDS <louise.a.clark@omb.eop.gov>
Co-authored-by: Rachel Lanman <rachellanman@Rachels-MacBook-Pro.local>
Co-authored-by: Rachel Lanman <Rachel.S.Lanman@omb.eop.gov>
Co-authored-by: rachellanman <121968284+rachellanman@users.noreply.github.com>
Co-authored-by: louiseClarkUSDS <111302353+louiseClarkUSDS@users.noreply.github.com>
Co-authored-by: SteveMattinglyUSDS <SteveMattinglyUSDS@users.noreply.github.com>

louiseClarkUSDS committed:
update cypress commands and adds comments

louiseClarkUSDS committed:
Merge branch 'staging' into 723-cypress-axe-wcag-testing

louiseClarkUSDS committed:
adds wait to make sure page is loaded before checks
